libraries/chain_kv/include/b1/chain_kv/chain_kv.hpp

```diff
@@ -130,22 +131,27 @@ namespace detail {
 
    template<typename T>
    struct value_storage<T*> {
-      constexpr char* as_char_ptr(T* value) const { return reinterpret_cast<char*>(value); }
-      constexpr std::size_t size() const { return sizeof(T); }
+      using type = T;
+      constexpr T* as_ptr(T* value) const { return (value); }
    };
 
    template<typename T>
    struct value_storage {
-      constexpr char* as_char_ptr(T& value) const { return reinterpret_cast<char*>(&value); }
-      constexpr std::size_t size() const { return sizeof(T); }
+      using type = T;
+      constexpr T* as_ptr(T& value) const { return &value; }
    };
 
    template <typename T, std::size_t N>
    auto append_key(bytes& dest, T value) {
-      constexpr static auto type_size = value_storage<T>().size();
-      constexpr static auto buf_size = type_size * N;
+
+      using t_type = typename value_storage<T>::type;
+      constexpr static auto buf_size = sizeof(t_type) * N;
       char buf[buf_size];
-      detail::copy_and_swap_to_buffer<type_size, N>(buf, value_storage<T>().as_char_ptr(value));
+      t_type* buf_as_type = reinterpret_cast<t_type*>(buf);
+      const t_type* first = value_storage<T>().as_ptr(value);
+      const t_type* last = first + N;
+      std::reverse_copy(first, last, buf_as_type);
```

Comment:
An array of `char` is not guaranteed to be correctly aligned to hold objects of type `t_type`.

Reply:
@swatanabe-b1 Are you saying that this is not guaranteed to work for uint64_t, uint128_t, key256_t, float64_t, and float128_t or are you saying that it isn't guaranteed to work for an arbitrary type_t?

Reply:
@swatanabe-b1 Huang-Ming helped me, I'll push the fix in a few minutes